src/Adapter/Product/QueryHandler/GetProductForEditingHandler.php

```diff
@@ -185,6 +202,8 @@ public function __construct(
         $this->productImageUrlFactory = $productImageUrlFactory;
         $this->specificPriceRepository = $specificPriceRepository;
         $this->configuration = $configuration;
+        $this->contextLangId = $contextLangId;
```

Comment:
On second thought, I'm sorry I have a lot of second thoughts in this review 😅, it would be interesting to move the `languageId` inside the query itself. It means the contextLangId would be injected in the `ProductFormDataProvider` to create the query object (and actually it already has it)

And it would allow behat testing this query with different languages. SO I would suggest adding a `$displayLanguageId` parameter to the query

And it's usually better when we can avoid injecting contextual values in the handler (when it's possible easily)